src/Esprima/JavascriptParser.cs

```diff
@@ -5387,17 +5475,26 @@ private ParserException UnexpectedTokenError(in Token token, string? message = n
     }
 
     [DoesNotReturn]
-    private protected void ThrowUnexpectedToken(in Token token = default, string? message = null)
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private protected void ThrowUnexpectedToken(object? boxedToken)
     {
-        throw UnexpectedTokenError(token, message);
+        throw UnexpectedTokenError(boxedToken is not null ? (Token) boxedToken : default);
     }
```

Comment:
Would be unnecessary if we used `StrongBox<T>`

Reply:
Now using stronger box